src/server/scripts/Pandaria/zone_the_wandering_isle.cpp

```diff
@@ -0,0 +1,2356 @@
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_TEMPERED_FURY:
+                        DoCastVictim(SPELL_TEMPERED_FURY);
+                        _events.ScheduleEvent(EVENT_TEMPERED_FURY, urand(2000, 4000));
+                        break;
+                    case EVENT_COMBAT_ROLL:
+                        // todo: cast combat roll only if it won't kick Vordraka outside of ship boundaries
+                        DoCastVictim(SPELL_COMBAT_ROLL);
+                        _events.ScheduleEvent(EVENT_TEMPERED_FURY, urand(5000, 7000));
+                        break;
+                    case EVENT_UPDATE_PHASES:
+                        std::list<Player*> players;
+                        me->GetPlayerListInGrid(players, me->GetVisibilityRange());
+                        for (std::list<Player*>::const_iterator itr = players.begin(); itr != players.end(); ++itr)
+                        {
+                            if ((*itr)->ToPlayer()->GetQuestStatus(QUEST_AN_ANCIENT_EVIL) == QUEST_STATUS_COMPLETE)
+                            {
+                                PhasingHandler::AddPhase(*itr, DB_PHASE_AFTER_FIGHT, true);
+                                PhasingHandler::RemovePhase(*itr, DB_PHASE_FIGHT, true);
+                            }
+                        }
+                        break;
+                }
+            }
+
+            DoMeleeAttackIfReady();
+        }
+
+    private:
+        EventMap _events;
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_aysa_vordraka_fightAI(creature);
+    }
+};
+
+class spell_tempered_fury : public SpellScript
+{
+    PrepareSpellScript(spell_tempered_fury);
+
+    void HandleJumpDest(SpellEffIndex effIndex)
+    {
+        PreventHitDefaultEffect(effIndex);
+        Position const jumpPos = GetHitDest()->GetPosition();
+        GetCaster()->GetMotionMaster()->MoveJump(jumpPos, 20, 0);
+    }
+
+    void Register() override
+    {
+        OnEffectLaunch += SpellEffectFn(spell_tempered_fury::HandleJumpDest, EFFECT_0, SPELL_EFFECT_JUMP_DEST);
+    }
+};
+
+enum VordrakaEvents
+{
+    EVENT_SMASH                     = 1,
+    EVENT_RUPTURE                   = 2,
+    EVENT_SUMMON_AGGRESSOR          = 3
+};
+
+enum VordrakaSpells
+{
+    SPELL_DEEP_SEA_RUPTURE          = 117456,
+    SPELL_FORCECAST_AGGRESSOR       = 117403,
+    SPELL_DEATH_INVIS               = 117555,
+    SPELL_SEE_DEATH_INVIS           = 117491
+};
+
+enum VordrakaMisc
+{
+    NPC_AYSA_CLOUDSINGER_VORDRAKA   = 56416,
+    NPC_DEEPSCALE_AGGRESSOR         = 60685
+};
+
+class npc_vordraka : public CreatureScript
+{
+public:
+    npc_vordraka() : CreatureScript("npc_vordraka") { }
+
+    struct npc_vordrakaAI : public ScriptedAI
+
```

Comment:
You can avoid grid lookups by using a SummonList + JustSummoned. Just because it's done in BossAI doesn't mean you can't reuse that construct here.

Reply:
But these NPCs that are summoned, are not summoned by Vordraka. He only casts spell ([117403](https://www.wowhead.com/spell=117403/forcecast-summon-deep-sea-aggressor)) on nearby player(s) and each player summons 2 NPS. Also I am not doing anything with these summoned creatures in this AI, only incrementing the variable value based on number of method calls.
I can replace the `uint8 _aggressorSummoned` with two booleans if its more likeable.

Reply:
oh ok, perks of not having spellwork i guess. leave it imho